server/src/test/java/org/elasticsearch/action/ActionModuleTests.java

```diff
@@ -196,4 +198,39 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
             threadPool.shutdown();
         }
     }
+
+    public void testCustomRestWrapperDeprecationMessage() {
+        class FakeHandler implements RestHandler {
+            @Override
+            public List<Route> routes() {
+                return singletonList(new Route(GET, "/_dummy"));
+            }
+
+            @Override
+            public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+            }
+        }
+        ActionPlugin secPlugin = new ActionPlugin() {
+            @Override
+            public UnaryOperator<RestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
+                return handler -> new FakeHandler();
+            }
+        };
+
+        SettingsModule moduleSettings = new SettingsModule(Settings.EMPTY);
+
+        ThreadPool threadPool = new TestThreadPool("testCustomRestWrapperDeprecationMessage");
+        try {
+            UsageService usageService = new UsageService();
+            new ActionModule(false, moduleSettings.getSettings(),
+                TestIndexNameExpressionResolver.newInstance(),
+                moduleSettings.getIndexScopedSettings(), moduleSettings.getClusterSettings(), moduleSettings.getSettingsFilter(),
+                threadPool, singletonList(secPlugin), null, null, usageService, null);
+            assertWarnings("The org.elasticsearch.action.ActionModuleTests$7 plugin installs a custom REST wrapper. " +
```

Comment:
Can we please use a named class rather than an anonymous class so that the message doesn't reference compiler-generated class names.

Reply:
It doesn't look like this has changed. The plugin is still an anonymous class, so it gets a synthetic name (`$7`) instead of a real name. That means that if another test gets changed, and this plugin isn't the 7th anonymous class anymore, then the test starts to fail because the message doesn't match the synthetic name.

Reply:
Sorry, forgot to commit half of the change －ლ